### ISSUE TITLE
Menus: auto select an item if it has a (1) badge

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#170701]
If a user has an update, or needs to login we show a badge on the activity bar icon. And we also update that menu entry to have `(1)` as the suffix.
When the user opens the menu with a badge - the item with the (1) suffix should be pre-selected.
If there are multiple, we should pre-select the first one.

I am not sure if this is possible with native menus.
Assigning first to @sbatten since I know he owns the menus

Slightly related - can we right align the (1) so it is easier to see?

Current behaviour on click:
![Screenshot 2023-01-06 at 17 13 04](https://user-images.githubusercontent.com/1926584/211052012-c5297221-eed4-4c31-b494-1f7e3a6ed50c.png)

What I would expect on click:
![Screenshot 2023-01-06 at 17 12 42](https://user-images.githubusercontent.com/1926584/211051944-312dc0c0-777d-4f49-96ec-5207257fa17c.png)

fyi @sandy081

**Comment by sbatten** [comment]
In native, the closest thing we can do is position the menu so the specific item is under the mouse cursor, but I would avoid this on Linux and Windows in the native case as there is a longstanding issue with the menus auto clicking an item on pop-up

**Comment by isidorn** [comment]
Ok, thanks. Makes sense for native menus.
But for custom menus - does it make sense to do this?

Also what do you think about right aligning the `(1)`? Is it possible to right align - do we know the character size of the area?

**Comment by sbatten** [comment]
I don't think we could support that in native either. We could do all of it in custom, though I've never seen the patter of pre-selecting an item in the menu unless it is at the top.

What about using an icon instead of a text based (1)? Something that looks similar to the badge?

**Comment by isidorn** [comment]
@sbatten if you think it goes against well known patterns then that is a good enough reason to NOT do it for me.

As for the icon - yeah I absolutely love the idea. @daviddossett 
But for native menus we can not do an icon. Would it make sense to right align in that case?

As an example here's how Apple does it:
![Screenshot 2023-01-06 at 18 49 31](https://user-images.githubusercontent.com/1926584/211069502-50b8f0f7-7d25-4eab-a269-fe1fa076e05e.png)

**Comment by sbatten** [comment]
We can put emoji in native menus, and we can also create icons from base64 encoded images

e.g. 

![image](https://user-images.githubusercontent.com/6561887/211078994-40796163-fd2b-40bd-9c59-3cb72755f10e.png)

**Comment by bpasero** [comment]
I am not sure I would think such a behaviour is expected, at least when the element is not the first. Also keep in mind users with accessibility needs where it maybe unexpected that some seamingly random entry is selected by default which changes often.

**Comment by bpasero** [comment]
I really like the "1 update" design in the native menu but I am not aware of a way to do this in Electron currently, so it may require an upstream change.

//cc @deepak1556 @rzhao271

**Comment by deepak1556** [comment]
Menu items `icon` property accepts input from [NativeImage](https://github.com/electron/electron/blob/main/docs/api/native-image.md) module which can generate from a wide variety of inputs, I can think of creating dynamic svg and using the bitmap as input for this particular case https://github.com/electron/electron/blob/main/docs/api/native-image.md#nativeimagecreatefrombitmapbuffer-options . However aligning the icon is not configurable which will require an upstream change.

**Comment by sbatten** [comment]
My image above is actually done in electron the way @deepak1556 mentions. So yes, we can actually do it dynamically and match the color if that looks good.

**Comment by bpasero** [comment]
Awesome 👏

**Comment by bpasero** [comment]
@sbatten given this, I wonder if we could tackle https://github.com/microsoft/vscode/issues/161347 even?

**Comment by isidorn** [comment]
Your image looks great. I think it should just be on the right side - not on the left.
That way the menu entries will be aligned with text start.
What do you think?

**Comment by sbatten** [comment]
I agree it would look better aligned to the right but since that is not an option today, I wonder if we want to wait for the upstream work before doing anything?

**Comment by isidorn** [comment]
Maybe we start so it is at the end of the text? And once upstream supports right align we just move it there. So it will be a "natural" transition.

**Comment by sbatten** [comment]
This icon is a separate property, so its position relative to the text is not currently configurable in any way, not even before or after text.

**Comment by esonnino** [comment]
There are context menus that have checks on them, could we use this space to represent the dot as a codicon? 
(See the Sync this extension item bellow) 
![image](https://user-images.githubusercontent.com/113370447/212194034-bc11d9f3-0eca-45cf-a4e6-4aefa91f7623.png)
![image](https://user-images.githubusercontent.com/113370447/212196239-43fe6e8f-43ca-41c0-b59b-377d5e5be365.png)

**Example:**
![image](https://user-images.githubusercontent.com/113370447/212197494-db64b9a7-401d-4624-9ccd-7b5b44b3aa8c.png)

Other visual explorations: 

![image](https://user-images.githubusercontent.com/113370447/212198272-8dbe75de-b883-49d3-aba5-b286b4bea217.png)

**Comment by sbatten** [comment]
@esonnino for custom menus its possible, but still not something we can do with electron api for native menus

**Comment by esonnino** [comment]
@sbatten, can we make the account menu custom? Which menus are native which ones are custom? :)

**Comment by daviddossett** [comment]
@esonnino all of the context menus use native UI, at least in macOS. vscode.dev, Windows, and Linux use custom menus.

<img width="363" alt="CleanShot 2023-01-13 at 09 47 57@2x" src="https://user-images.githubusercontent.com/25163139/212386285-325aaae9-1593-4553-a2a1-cb516f3b3223.png">

**Comment by sbatten** [comment]
small detail: by default linux uses native as well, but can use custom like Windows.
